irohad/ametsuchi/impl/postgres_specific_query_executor.cpp

```diff
@@ -227,7 +227,7 @@ namespace iroha {
         }
         auto &tx = block->transactions()[tx_id];
         if (pred(tx)) {
-          *dest_it++ = clone(tx);
+          *dest_it++ = tx.moveTo();
```

Comment:
would not it be cleaner if we add a non-const getter to block txs?
```c++
        auto &tx = block->mutableTransactions()[tx_id];
        if (pred(tx)) {
          *dest_it++ = std::move(tx);
        }
```
and
```c++
      using MutableTransactionsCollectionType =
          boost::any_range<Transaction &,
                           boost::random_access_traversal_tag,
                           Transaction &>;
      MutableTransactionsCollectionType Block::transactions();
```

Reply:
In `*dest_it++ = std::move(tx);` std::move will not work. Because the type in container is abstract interface::Transaction. Compiler doesn't know what kind of type to construct.